packages/backend/src/ee/services/McpService/McpService.ts

```diff
@@ -0,0 +1,54 @@
+// eslint-disable-next-line import/extensions
+import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
+import { LightdashConfig } from '../../../config/parseConfig';
+import { VERSION } from '../../../version';
+
+export enum McpToolName {
+    GET_LIGHTDASH_VERSION = 'get_lightdash_version',
+}
+
+type McpServiceArguments = {
+    lightdashConfig: LightdashConfig;
+};
+
+export class McpService {
+    private lightdashConfig: LightdashConfig;
+
+    private mcpServer: McpServer;
+
+    constructor({ lightdashConfig }: McpServiceArguments) {
+        this.lightdashConfig = lightdashConfig;
+        try {
+            this.mcpServer = new McpServer({
+                name: 'Lightdash MCP Server',
+                version: VERSION,
+            });
+            this.setupHandlers();
+        } catch (error) {
+            console.error('Error initializing MCP server:', error);
+            throw error;
+        }
+    }
+
+    private setupHandlers(): void {
+        this.mcpServer.registerTool(
+            McpToolName.GET_LIGHTDASH_VERSION,
+            {
+                description: 'Get the current Lightdash version',
+                inputSchema: {},
+            },
+            async () => ({
+                content: [{ type: 'text', text: this.getLightdashVersion() }],
+            }),
+        );
+    }
+
+    public getServer(): McpServer {
+        return this.mcpServer;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    public getLightdashVersion(): string {
```

Comment:
lets add at least one permission check here:
- The instance should have an environment variable set to enable MCP
- The user (token) should have the CASL ability (oauth scope) for `mcp`